File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        
        HashMap<Character, Character> myMap = new HashMap();
        myMap.put(')','(');
        myMap.put('}','{');
        myMap.put(']','[');
        
        Stack<Character> myStack = new Stack();
        
        for(int i=0; i<s.length(); i++){
            char curr = s.charAt(i);
            if(myMap.containsKey(curr)){
                char pop = myStack.size() != 0 ? myStack.pop() : '#';
                if(pop != myMap.get(curr)){
                    return false;
                }
                
            }else{
                myStack.push(curr);
            }
            
        }
        
        return myStack.isEmpty();
        
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    if(s.length % 2 === 1) return false;
    const pairs = {
        ')' : '(',
        '}' : '{',
        ']' : '['
    }
    const myStack = [];
    for(var i=0; i<s.length; i++){
        if(s[i] === '(' || s[i] === '{' || s[i] === '['){
            myStack.push(s[i]);
        }else{ // closed parentheses ([]) myStack (   
            if( myStack.pop() !== pairs[s[i]]){
                return false;
            }
        }
    }
    return myStack.length === 0;
};